radosgw_agent/worker.py

```diff
@@ -99,15 +101,65 @@ def _meta_entry_from_json(entry):
         )
 
 BucketIndexEntry = namedtuple('BucketIndexEntry',
-                              ['object', 'marker', 'timestamp'])
+                              [
+                                  'object',
+                                  'marker',
+                                  'timestamp',
+                                  'op',
+                                  'versioned',
+                                  'ver',
+                                  'name',
+                                  # compatibility with boto objects:
+                                  'VersionedEpoch',
+                                  'version_id',
+                              ])
+
+BucketVer = namedtuple('BucketVer',
+        [
+            'epoch',
+            'pool',
+        ])
+
 
 def _bi_entry_from_json(entry):
+    ver = entry.get('ver', {})
+    entry_ver = BucketVer(
+        ver.get('epoch'),
+        ver.get('pool')
+    )
+
+    # compatibility with boto objects:
+    VersionedEpoch = ver.get('epoch')
+    version_id = entry.get('instance')
```

Comment:
I just put the design doc into an etherpad for easier reference: http://pad.ceph.com/p/radosgw-agent-object-versioning